Support for success notices.

jonathanmarvens opened the pull request:
The **2.0.0** package version update is necessary, because this is a breaking update.

(Also updated **README.md** to reflect the changes.)

jonathanmarvens committed:
Added support for success notices!

jonathanmarvens committed:
Updated the tests to support success notices.

jonathanmarvens committed:
Updated the documentation to reflect the support for success notices.

jonathanmarvens committed:
v2.0.0.

jonathanmarvens commented:
**CC:** @jefflembeck.

aredridel commented:
This looks great! The only thing I'd add is to filter out empty notices -- this can simplify the use of the module because you can `doComplicatedThing().then(x => null).catch(e => throw "It went bad!!!")` and put that promise in, success or failure but only notifying on failure.

jefflembeck commented:
I'm diggin this a _lot_.

jefflembeck committed:
This ignores empty notices

Addresses: https://github.com/npm/hapi-stateless-notifications/pull/5#issuecomment-173615999

bcoe commented on index.js:
```diff
@@ -47,7 +59,22 @@ exports.register = function(server, options, next) {
                 request.response.source.context = {};
             }
 
-            request.response.source.context.notices = data.notices;
```
there's enough logic sneaking in here that I'd be tempted to move it into a helper function that `server.ext('onPreResponse')` invokes.

`function assignNoticesToContext () ???`

bcoe commented on test.js:
```diff
@@ -69,7 +72,12 @@ test('does it work?', function (t) {
             t.ok(token);
             server.inject({ method: "GET", url: '/2?notice=' + token}, function (res) {
-                t.equal(res.result.trim(), 'notice: boom');
+                var renderedNotices = res.result.trim().split('\n').map(function(value) {
```
As an outsider coming in I'd love to see the tests split into a few assertions, rather than everything being contained in the top-level description, `does it work?` perhaps we could have something like this:

``` js
describe('hapi-stateless-notifications', function () {
  it('handles success notices', function () {

  })

  it('handles error notices', function () {

  })

  it('handles a mix of success and error notifications', function () {

  })

  it('handles no notices in the session', function () {

  })
})
```

Getting this foundation in place will make it super easy for other folks to swoop in and add functionality \o/

stubbornella replied:
+1

bcoe commented:
@jonathanmarvens this is looking great, really love how you've extended this module to solve a more general case :+1: 

My main comments are about tests, seeing tests around the various permutations of this library would help my confidence -- especially as someone not as familiar with front-end-code as others.

jefflembeck commented:
I'm going to merge this and create an issue to address @bcoe's changes so they can be addressed in the next patch.
